Dockerfile

```diff
@@ -102,18 +101,45 @@ RUN PKG_CONFIG="/usr/bin/$(xx-info)-pkg-config" \
     && mv target/$(xx-cargo --print-target-triple)/$PROFILE_DIR/qdrant /qdrant/qdrant
 
 # Download and extract web UI
-RUN mkdir /static ; STATIC_DIR='/static' ./tools/sync-web-ui.sh
+RUN mkdir /static && STATIC_DIR=/static ./tools/sync-web-ui.sh
 
 
 FROM debian:12-slim AS qdrant
 
-RUN apt-get update \
-    && apt-get install --no-install-recommends -y ca-certificates tzdata libunwind8 \
+RUN apt-get update
+
+# Install additional packages into the container.
+# E.g., the debugger of choice: gdb/gdbserver/lldb.
+ARG PACKAGES
+
+RUN apt-get install -y --no-install-recommends ca-certificates tzdata libunwind8 $PACKAGES \
     && rm -rf /var/lib/apt/lists/*
 
-ARG APP=/qdrant
+# Copy Qdrant source files into the container. Useful for debugging.
+#
+# To enable, set `SOURCES` to *any* non-empty string. E.g., 1/true/enable/whatever.
+# (Note, that *any* non-empty string would work, so 0/false/disable would enable the option as well.)
+ARG SOURCES
 
-RUN mkdir -p "$APP"
+# Dockerfile does not support conditional `COPY` instructions (e.g., it's impossible to do something
+# like `if [ -n "$SOURCES" ]; then COPY ...; fi`), so we *hack* conditional `COPY` by abusing
+# parameter expansion and `COPY` wildcards support. 😎
```

Comment:
If source path of a `COPY` instruction uses a wildcard, and the wildcard does not match anything, then `COPY` instruction is silently ignored without producing an error.

If `SOURCES` build-arg is not set (or set to an empty string), then each `${DIR:-/null?}` would expand to `/null?` (which contains `?` wildcard!), and because there's no path matching `/null?` in the `builder` step, each `COPY` command would be silently ignored.

(And, of course, if `SOURCES` arg is set to *anything*, then each `${DIR:-/null?}` expansion would expand to the directory path and be copied properly.)